src/RRule.php

```diff
@@ -2324,4 +2339,27 @@ public function humanReadable(array $opt = array())
 		return $str;
 	}
 
+	/**
+	 * Attempt to fetch a locale either from Intl or setlocale()
+	 *
+	 * @return null|string
+	 */
+	private function getSystemLocale()
+	{
+		$locale = null;
+
+		if ( self::$intl_loaded ) {
+			$locale = \Locale::getDefault();
+		} else {
+			$locales = setlocale(LC_ALL, 0);
```

Comment:
You should use `setlocale(LC_MESSAGES, 0)`, so you get the locale right away and you don't need to do a `preg_match`.